web_ui/prometheus_test.go

```diff
@@ -139,206 +144,133 @@ func TestPrometheusProtectionOriginHeaderScope(t *testing.T) {
+		w := httptest.NewRecorder()
+		c, r := gin.CreateTestContext(w)
+		c.Request = &http.Request{
+			URL: &url.URL{},
+		}
 
-	// Set the request to go through the promQueryEngineAuthHandler function
-	r.GET("/api/v1.0/prometheus/*any", promQueryEngineAuthHandler(av1))
-	c.Request, _ = http.NewRequest(http.MethodGet, "/api/v1.0/prometheus/test", bytes.NewBuffer([]byte(`{}`)))
+		// Load the private key
+		privKey, err := config.GetIssuerPrivateJWK()
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	// Put the signed token within the header
-	c.Request.Header.Set("Authorization", "Bearer "+string(signed))
-	c.Request.Header.Set("Content-Type", "application/json")
+		token := createToken("monitoring.query", issuerUrl, privKey)
 
-	r.ServeHTTP(w, c.Request)
+		// Set the request to go through the promQueryEngineAuthHandler function
+		r.GET("/api/v1.0/prometheus/*any", promQueryEngineAuthHandler(av1))
+		c.Request, _ = http.NewRequest(http.MethodGet, "/api/v1.0/prometheus/test", bytes.NewBuffer([]byte(`{}`)))
 
-	assert.Equal(t, 404, w.Result().StatusCode, "Expected status code of 404 representing failure due to minimal server setup, not token check")
+		// Put the signed token within the header
+		c.Request.Header.Set("Authorization", "Bearer "+string(token))
+		c.Request.Header.Set("Content-Type", "application/json")
 
-	// Create a new Recorder and Context for the next HTTPtest call
-	w = httptest.NewRecorder()
-	c, r = gin.CreateTestContext(w)
+		r.ServeHTTP(w, c.Request)
 
-	c.Request = &http.Request{
-		URL: &url.URL{},
-	}
+		assert.Equal(t, 404, w.Result().StatusCode, "Expected status code of 404 representing failure due to minimal server setup, not token check")
 
-	// Create a private key to use for the test
-	privateKey, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
-	assert.NoError(t, err, "Error generating private key")
+	})
 
-	// Convert from raw ecdsa to jwk.Key
-	pKey, err := jwk.FromRaw(privateKey)
-	assert.NoError(t, err, "Unable to convert ecdsa.PrivateKey to jwk.Key")
+	t.Run("invalid-token-sig-key", func(t *testing.T) {
+		// Create a new Recorder and Context for the next HTTPtest call
+		w := httptest.NewRecorder()
+		c, r := gin.CreateTestContext(w)
 
-	//Assign Key id to the private key
-	err = jwk.AssignKeyID(pKey)
-	assert.NoError(t, err, "Error assigning kid to private key")
+		c.Request = &http.Request{
+			URL: &url.URL{},
+		}
 
-	//Set an algorithm for the key
-	err = pKey.Set(jwk.AlgorithmKey, jwa.ES256)
-	assert.NoError(t, err, "Unable to set algorithm for pKey")
+		// Create a private key to use for the test
+		privateKey, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
+		assert.NoError(t, err, "Error generating private key")
 
-	jti_bytes = make([]byte, 16)
-	_, err = rand.Read(jti_bytes)
-	if err != nil {
-		t.Fatal(err)
-	}
-	jti = base64.RawURLEncoding.EncodeToString(jti_bytes)
+		// Convert from raw ecdsa to jwk.Key
+		pKey, err := jwk.FromRaw(privateKey)
```

Comment:
How about we change to `privKey` for clarity, since `pkey` could stand for either public or private.